lms/djangoapps/instructor/static/instructor/ReportPopup/components/Popup/PopupModal.jsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import {Modal, Button, Hyperlink} from '@edx/paragon/static';
+
+export default class PopupModal extends React.Component {
+    constructor(props) {
+        super(props);
+        this.handleDownload = this.handleDownload.bind(this);
+    }
+
+    handleDownload() {
+        window.open(`/media/${this.props.reportPath}`, '_blank');
+    }
+
+    render() {
+        let buttons = [];
+
+        if (this.props.reportPath != null) {
+            buttons.push((
+                <Button label={gettext("Download CSV")} onClick={this.handleDownload}/>
+            ));
+        }
+
+        let progress = null;
+        if (this.props.reportPath == null) {
+            if (this.props.inProgress) {
+                progress = (
+                    <div class="title">
+                        <b>{gettext("Your report is being created...")}</b>
+                    </div>
+                );
+            }
+        }
+
+        let msg = null;
+        if (this.props.error != null) {
+            msg = (
+                <div className="msg msg-error">{this.props.error}</div>
+            );
+        } else if (this.props.message != null) {
+            msg = (
+                <div
+                    className={`msg ${this.props.succeeded ? "msg-success" : "msg-warning"}`}>{this.props.message}</div>
+            );
+        }
+
+        const body = (
+            <div className={"report-popup-modal-body"}>
+                {progress}
+                {msg}
+                <p>
+                    Once it's ready, you can view or download it using the buttons below. You can also close this popup
+                    now, and download the report later, from the "Reports Available for Download" area below.
+                </p>
+            </div>
+        );
+
+        return (
+            <Modal title={gettext("Learner Response Report")}
+                   body={body}
+                   buttons={buttons}
+                   onClose={this.props.onHide}
+                   open={this.props.open}
+                   closeText={gettext("Close")} />
+        );
+    }
+}
+
+PopupModal.propTypes = {
+    open: PropTypes.bool,
+    onHide: PropTypes.func.isRequired,
+
+    message: PropTypes.string,
+    error: PropTypes.string,
+    inProgress: PropTypes.bool.isRequired,
+    succeeded: PropTypes.bool.isRequired,
+    reportPath: PropTypes.string,
+};
```

Comment:
This can be a static functional component.

Reply:
done